lib/cache_mngr.c

```diff
@@ -148,6 +148,33 @@ static int event_input(struct nl_msg *msg, void *arg)
  */
 int nl_cache_mngr_alloc(struct nl_sock *sk, int protocol, int flags,
 			struct nl_cache_mngr **result)
+{
+	struct nl_sock *sk_blocking = nl_socket_alloc();
+	int err;
+
+	if (!sk_blocking) {
+		return -NLE_NOMEM;
+	}
+	if ((err = nl_connect(sk_blocking, protocol)) < 0) {
+			nl_socket_free(sk_blocking);
+			return err;
+	}
+	return nl_cache_mngr_alloc_ex(sk, sk_blocking, protocol, flags, result);
```

Comment:
as said, the logic of `nl_connect()` should not be on the caller.

Hence, most of this code is still in `nl_cache_mngr_alloc_ex()`.

Allow NULL as `sk_blocking` (`sync_sk`), and then just do
```C
int nl_cache_mngr_alloc(struct nl_sock *sk, int protocol, int flags,
			struct nl_cache_mngr **result)
{
    return nl_cache_mngr_alloc_ex(sk, NULL, protocol, flags, result);
}
```

Reply:
That was how I made it originally. However I noticed that presents a problem: we allow passing in two sockets, but cannot differentiate between errors caused by either. Which means we would need extra steps to be taken, to disown them internally in case of error, so that caller could release these.
I just found it easier to be inconsistent with socket assumptions.
I can fix the assumption about the second socket as well to be consistent.